pkg/controller/appdefinition/networkpolicy.go

```diff
@@ -0,0 +1,187 @@
+	podCIDR := *cfg.PodCIDR
+	// make sure the podCIDR is valid
+	if podCIDR != "" {
+		_, _, err = net.ParseCIDR(podCIDR)
+		if err != nil {
+			return err
+		}
+	}
+
+	// next, create NetworkPolicies for each container and job in the app that has a published port
+	for containerName, container := range app.Status.AppSpec.Containers {
+		buildNetPolsForContainer(app.Name, containerName, podNamespace, ingressNamespace, podCIDR, container, resp)
+	}
+	for jobName, job := range app.Status.AppSpec.Jobs {
+		buildNetPolsForContainer(app.Name, jobName, podNamespace, ingressNamespace, podCIDR, job, resp)
+	}
+
+	return nil
+}
+
+func buildNetPolsForContainer(appName, containerName, podNamespace, ingressNamespace, podCIDR string, container v1.Container, resp router.Response) {
+	for _, port := range container.Ports {
+		if port.Publish {
+			if port.Protocol == v1.ProtocolHTTP {
+				resp.Objects(buildNetPolForHTTPPublishedPort(
+					fmt.Sprintf("%s-%s-%s", strings.ToLower(appName), strings.ToLower(containerName), strconv.Itoa(int(port.Port))),
+					podNamespace, ingressNamespace, containerName, port.Port))
+			} else {
+				resp.Objects(buildNetPolForOtherPublishedPort(
+					fmt.Sprintf("%s-%s-%s", strings.ToLower(appName), strings.ToLower(containerName), strconv.Itoa(int(port.Port))),
+					podNamespace, podCIDR, containerName, port.Port))
+			}
+		}
+	}
+	// create policies for the sidecars as well
+	for sidecarName, sidecar := range container.Sidecars {
+		for _, port := range sidecar.Ports {
+			if port.Publish {
+				if port.Protocol == v1.ProtocolHTTP {
+					resp.Objects(buildNetPolForHTTPPublishedPort(
+						fmt.Sprintf("%s-%s-sidecar-%s-%s", strings.ToLower(appName), strings.ToLower(containerName), strings.ToLower(sidecarName), strconv.Itoa(int(port.Port))),
+						podNamespace, ingressNamespace, containerName, port.Port))
+				} else {
+					resp.Objects(buildNetPolForOtherPublishedPort(
+						fmt.Sprintf("%s-%s-sidecar-%s-%s", strings.ToLower(appName), strings.ToLower(containerName), strings.ToLower(sidecarName), strconv.Itoa(int(port.Port))),
+						podNamespace, podCIDR, containerName, port.Port))
+				}
+			}
+		}
+	}
+}
+
+func buildNetPolForHTTPPublishedPort(name, namespace, ingressNamespace, containerName string, port int32) *networkingv1.NetworkPolicy {
+	var namespaceSelector metav1.LabelSelector
+	if ingressNamespace != "" {
+		namespaceSelector = metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"kubernetes.io/metadata.name": ingressNamespace,
+			},
+		}
+	}
+
+	return &networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					labels.AcornContainerName: containerName,
+				},
+			},
+			Ingress: []networkingv1.NetworkPolicyIngressRule{{
+				From: []networkingv1.NetworkPolicyPeer{{
+					NamespaceSelector: &namespaceSelector,
```

Comment:
When `ingressNamespace` is not set, does an empty LabelSelector mean matching to every namespace? I thought you have to use `labels.everything()`

Reply:
Yep, empty means everything in this particular case.